build_projects/dotnet-host-build/MsiTargets.cs

```diff
@@ -0,0 +1,185 @@
+
+            if (Directory.Exists(wixObjRoot))
+            {
+                Utils.DeleteDirectory(wixObjRoot);
+            }
+            Directory.CreateDirectory(wixObjRoot);
+
+            Cmd("powershell", "-NoProfile", "-NoLogo",
+                Path.Combine(Dirs.RepoRoot, "packaging", "windows", "host", "generatemsi.ps1"),
+                inputDir, SharedHostMsi, WixRoot, sharedHostBrandName, hostMsiVersion, hostNugetVersion, Arch, wixObjRoot)
+                    .Execute()
+                    .EnsureSuccessful();
+            return c.Success();
+        }
+
+        [Target]
+        [BuildPlatforms(BuildPlatform.Windows)]
+        public static BuildTargetResult GenerateDotnetSharedFrameworkMsi(BuildTargetContext c)
+        {
+            var inputDir = c.BuildContext.Get<string>("SharedFrameworkPublishRoot");
+            var sharedFrameworkNuGetName = Monikers.SharedFrameworkName;
+            var sharedFrameworkNuGetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
+            var msiVerison = sharedFrameworkNuGetVersion.Split('-')[0];
+            var upgradeCode = Utils.GenerateGuidFromName(SharedFrameworkMsi).ToString().ToUpper();
+            var wixObjRoot = Path.Combine(Dirs.Output, "obj", "wix", "sharedframework");
+            var sharedFxBrandName = $"'{Monikers.SharedFxBrandName}'";
+
+            if (Directory.Exists(wixObjRoot))
+            {
+                Utils.DeleteDirectory(wixObjRoot);
+            }
+            Directory.CreateDirectory(wixObjRoot);
+
+            Cmd("powershell", "-NoProfile", "-NoLogo",
+                Path.Combine(Dirs.RepoRoot, "packaging", "windows", "sharedframework", "generatemsi.ps1"),
+                inputDir, SharedFrameworkMsi, WixRoot, sharedFxBrandName, msiVerison, sharedFrameworkNuGetName, sharedFrameworkNuGetVersion, upgradeCode, Arch, wixObjRoot)
+                    .Execute()
+                    .EnsureSuccessful();
+            return c.Success();
+        }
+        
+        [Target(nameof(MsiTargets.InitMsi))]
+        [BuildPlatforms(BuildPlatform.Windows)]
+        public static BuildTargetResult GenerateSharedFxBundle(BuildTargetContext c)
+        {
+            var sharedFrameworkNuGetName = Monikers.SharedFrameworkName;
+            var sharedFrameworkNuGetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
+            var upgradeCode = Utils.GenerateGuidFromName(SharedFrameworkBundle).ToString().ToUpper();
+            var sharedFxBrandName = $"'{Monikers.SharedFxBrandName}'";
+
+            Cmd("powershell", "-NoProfile", "-NoLogo",
+                Path.Combine(Dirs.RepoRoot, "packaging", "windows", "sharedframework", "generatebundle.ps1"),
+                SharedFrameworkMsi, SharedHostMsi, SharedFrameworkBundle, WixRoot, sharedFxBrandName, MsiVersion, CliDisplayVersion, sharedFrameworkNuGetName, sharedFrameworkNuGetVersion, upgradeCode, Arch)
```

Comment:
Dug through this, `CLIDisplayVersion` get's passes as the version of the bundle, but it is a misnomer, it really represents the 4-part build number. This is just a rename.